string.c

```diff
@@ -3602,24 +3602,29 @@ str_uplus(VALUE str)
 
 /*
  * call-seq:
- *   -string -> frozen_string
- *   dedup -> frozen_string
+ *   -self -> frozen_string
  *
- * Returns a frozen, possibly pre-existing copy of the string.
+ * Returns a frozen string containing the same content as +self+.
  *
- * The returned +String+ will be deduplicated as long as it does not have
- * any instance variables set on it and is not a String subclass.
+ * The returned +frozen_string+ is +self+, frozen, if +self+:
```

Comment:
`self` is only returned if it was already frozen (and then the two conditions you mentioned).

Reply:
Recast.